internal/db/read_writer_test.go

```diff
@@ -3193,3 +3193,35 @@ func TestDb_oplogMsgsForItems(t *testing.T) {
 // 		})
 // 	}
 // }
+
+func TestDb_Now(t *testing.T) {
+	t.Parallel()
+	conn, _ := TestSetup(t, "postgres")
+	rw := New(conn)
+	ctx := context.Background()
+	now, err := rw.Now(ctx)
+	require.NoError(t, err)
+	// Check that it's within 1 second of now according to the system
+	// If this is flaky... just increase the limit 😬.
+	assert.True(t, now.Before(time.Now().Add(time.Second)))
+	assert.True(t, now.After(time.Now().Add(-time.Second)))
+
+	now2, err := rw.Now(ctx)
+	require.NoError(t, err)
+
+	// Calling Now() outside a transaction should result in different timestamps
+	assert.False(t, now.Equal(now2))
```

Comment:
`testify` [falls](https://github.com/stretchr/testify/blob/f97607b89807936ac4ff96748d766cf4b9711f78/assert/assertions.go#L801) [back to](https://github.com/stretchr/testify/blob/f97607b89807936ac4ff96748d766cf4b9711f78/assert/assertions.go#L65) [`reflect.DeepEqual`](https://pkg.go.dev/reflect#DeepEqual) for equality comparisons, and this is not a reliable way to compare timestamps for equality since it performs field by field comparisons. See https://stackoverflow.com/questions/75824869/reflect-deepequal-not-working-on-timestamp-comparison, https://wcsiu.com/2018/07/06/go-reflect-deep-equal-pitfall.html, etc. I make a practice of using `time.Equal` whenever I need to do timestamp comparisons, even in cases where it doesn't _really_ matter.

Reply:
I agree that using `reflect.DeepEqual` to compare instances of `time.Time` that have been created using different constructors from the `time` package can cause errors. However, in this case, all instances of `time.Time` are coming from the same lines of code which should produce instances that are equal using `reflect.DeepEqual`.

Either way, can you add a comment in the test code to capture the reasoning?

Reply:
Sure!